src/wasm.ts

```diff
@@ -0,0 +1,72 @@
+import { ExecuteResult, InstantiateResult, UploadResult } from "@cosmjs/cosmwasm-stargate";
+import { Coin } from "@cosmjs/proto-signing";
+import * as fs from "fs";
+import { ArchwaySigningClient, extractAddress } from "./connect";
+
+// Upload a contract to the  blockchain
+// Returns a receipt if contract successfully uploaded with Code ID
+export async function uploadContract(this: ArchwaySigningClient, wasmPath: string ): Promise<UploadResult> {
+    let wasm_file = fs.readFileSync(wasmPath);
+    let address = await extractAddress(this.wallet);
+    const fee = {
+        amount: [
+          {
+            denom: "uconst",
```

Comment:
also, `denom` will vary based on network

Reply:
True denom should differ after networks introduced.